packages/nimble-components/src/table/specs/data-hierarchy-hld.md

```diff
@@ -0,0 +1,199 @@
+
+<sup>1</sup>_The "leaf-only" mode mentioned in the 'Future work' section presents a caveat to this, in which parent rows are explicitly configured to not support certain actions, but we still may allow a user to perform selection through._
+
+### ARIA
+
+A parent row would have the same ARIA expectations of any child row, with the additional need to supply the `aria-expanded` attribute when it is expanded.
+
+Things to consider:
+
+-   Putting an appropriate label/title on the progress indicator for the lazy loaded row placeholder.
+-   Expand/collapse button attributes for the parent row. Likely should just match what is done for the group row.
+
+### Future work
+
+#### Leaf-only mode
+
+There are existing SLE grid variants (namely the Tags grid) that provide a type of hierarchical display that doesn't align exactly with either the proposed 'data hierarchy' outlined in this HLD, nor the grouping capability. Instead, while the parent rows may possibly be presented as a normal row (containing values in some subset of columns that the children have values for), they will not be able to be acted upon in the way child rows can, such as allowing an action menu to be defined on them, or being selected individually.
+
+It is expected that this mode will involve creating a new API on the table in order to fully enable, but it should be able to use the other APIs defined in this HLD to support it, and leverage capabilities such as lazy-loading.
+
+Here are a few of the considerations that were made with respect to this mode:
+
+-   grouping should group on leaf items, not parents (could be cumbersome to accomplish with Tanstack)
+-   action menu only on leafs
+-   single selection mode can only select leafs
+-   selection state (when set to `multiple`) of parents (if selected) should become indetermine when new rows loaded (e.g. via lazy loading)
+-   selection counts should ignore parents
+-   (maybe?) leaf-mode + multi-selection + lazy loading is considered an invalid confiuguration. This could mean that the Table API of `forceExpandCollapseFieldName` should be more semantically associated with lazy loading.
+    -   (alternative?): Could we instead just hide the selection checkbox for parent rows that have no children?
+
+#### Auto-expand-parent API
+
+Currently, the proposed default behavior of parent rows is that they will display as collapsed initially (unlike groups). This is primarily done to support the lazy-loading case where we never want a row that has no children to default to being expanded. We could add an API that allows a user to determine what the default state of parents should be, which would ultimately not apply to rows that have no children (i.e. rows with children could be auto expanded, but any parent rows without children would remain collapsed).
```

Comment:
I don't remember seeing the rationale for this being "future work" (but it's quite possible I just missed that discussion). I'm fairly certain this is something we need, so why not include it as part of the initial HLD? Are there open questions/concerns about this functionality that we are trying not to solve right now?

Reply:
I've now added to this to the proposed API section.